Components/CategoryResolver/AutoCategoryResolver.php

```diff
@@ -73,51 +73,47 @@ public function resolve(array $categories)
                 'name' => $node['name'],
                 'parentId' => 1,
             ]);
+            // if connectTree has a Subtree starting with Spanish but MerchantShop has no mainCategory Spanish
+            // the categories below Spanish won't be created
+            if ($mainCategory == null) {
+                continue;
+            }
 
-            $remoteCategories = array_merge($remoteCategories, $this->convertTreeToEntities($node['children'], $mainCategory));
+            $remoteCategories = $this->convertTreeToKeys($node['children'], $mainCategory->getId());
```

Comment:
AutoCategoryResolver still returns only leaf categories.
Won't be possible to use it and assign categories to categories which are not leaf.

Reply:
convert tree to keys has a bool parameter if only leafs will be returned 
per default it is set to true
the import service calls the convertToKeys method directly with this parameter set to false
the resolve function is just called in productToShop and there it is right to return just leafs